### PR TITLE
Popover click away

### PR DESCRIPTION
**Pull request by korbinancell**
- Add a new component ClickAwayHandler as a wrapper for [react-event-listener](https://www.npmjs.com/package/react-event-listener).
- Add the new component to the popover as well as a new prop to handle click away.

### CODE REVIEW

File: components/utils/click-away-handler.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ReactEventListener from 'react-event-listener';
+
+export class ClickAwayHandler extends Component {
+	static propTypes = {
+		/** Which mouse events should trigger the handler */
```
Comment (awesomebob) [inline components/utils/click-away-handler.jsx]: I'm not clear on what the use case is for picking a different mouseEvent or touchEvent would be.
If mouseDown and onTouchStart are the clear primary use-case, could we make those into defaultProps?

File: components/popover/component.jsx [hunk 2]
```diff
@@ -135,35 +138,42 @@ export class Popover extends React.Component {
...
-								...(!hideArrow ? Styled.margins[Styled.getPlacement(placement)] : {}),
-							}}
-							delay={delay}
-							onAnimationEnd={this.handleTransition}
-							hideArrow={hideArrow}
-							styleOverrides={styleOverrides}
+						<ClickAwayHandler
+							mouseEvent="onMouseDown"
+							touchEvent="onTouchStart"
+							{...(showPopper && onClickAway ? { onClick: onClickAway } : {})}
```
Comment (awesomebob) [inline components/popover/component.jsx]: I realize that you're being careful to *not* add an `onClick: null` prop, but is that necessary?
It looks to me that the ClickAwayHandler checks whether `onClick` is truthy before using it.

File: components/utils/click-away-handler.jsx (hunk 1 above)
Comment (korbinancell) [reply components/utils/click-away-handler.jsx]: I think there's a use case for wanting to listen to different events depending on the order of things. onMouseDown is before onBlur is before onClick. It has mattered before. I agree on the default props though,

Comment (awesomebob) [reply components/utils/click-away-handler.jsx]: Just need those defaultProps and I'm ready to merge.

**Pull request merged**